code/modules/vehicles/powerloader.dm

```diff
@@ -133,6 +176,60 @@
 	if(linked_powerloader.buckled_mob && linked_powerloader.buckled_mob == user)
 		linked_powerloader.unbuckle() //drop a clamp, you auto unbuckle from the powerloader.
 
+/obj/item/powerloader_clamp/update_icon()
+	var/icon_tag = ""
+	if(loaded)
+		if(istype(loaded, /obj/structure/dropship_equipment))
+			icon_tag = "ds_gear"
+
+		else if(istype(loaded, /obj/structure/ship_ammo))
+			if(istype(loaded, /obj/structure/ship_ammo/rocket))
+				icon_tag = "ds_rocket"
+			else
+				icon_tag = "ds_ammo"
+
+		else if(istype(loaded, /obj/structure/ob_ammo))
+			if(istype(loaded, /obj/structure/ob_ammo/warhead))
+				icon_tag = "ob_warhead"
+			else
+				icon_tag = "ob_fuel"
+
+		else if(istype(loaded, /obj/item/hardpoint/holder))
+			icon_tag = "vehicle_module"
+
+		else if(istype(loaded, /obj/structure/closet/crate))
+			icon_tag = "crate"
+
+		else if(istype(loaded, /obj/structure/reagent_dispensers))
+			icon_tag = "reagent_dispenser"
+
+		else if(istype(loaded, /obj/structure/machinery/floodlight))
+			icon_tag = "floodlight"
+
+		else if(istype(loaded, /obj/structure/bed/chair))
+			icon_tag = "chairs"
+
+		else
+			icon_tag = "big_crate"
+
+	if(!icon_tag)
+		icon_tag = "loader_clamp"
+
+	if(is_right)
+		icon_tag += "_right"
```

Comment:
this can all be majorly optimised

make a proc called handle_loaded_icon and make it return what will be the icon_tag of the object 

call that proc on the loaded item and make the icon_tag var the result, if it returns nothing then make it the default